From 3b92a099b5970b0fc4982630f1e753b4dcd03340 Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@inai.de>
Date: Thu, 24 Sep 2026 13:34:16 +0200
Subject: [PATCH] build: drop -D_TIME_BITS=64

libvmime is built without -D_TIME_BITS, and the
`vmime::datetime::datetime(time_t, int)` constructor gets encoded in
the ABI as `_ZN5vmime8datetimeC1Eli`. Gromox however builds with
-D_TIME_BITS=64, and the symbol reference becomes
`_ZN5vmime8datetimeC1Exi`.

```
  CXXLD    tests/bodyconv
/usr/lib/gcc/i586-suse-linux/16/../../../../i586-suse-linux/bin/ld:
./.libs/libgromox_mapi.so: undefined reference to
`vmime::datetime::datetime(long long, int)'
```

Drop the two preprocessor flags. They are the default mode for 64-bit
userspaces anyway, and 32-bit environment builders need to
consistently present flags (e.g. Debian injects -D_TIME_BITS=64
globally for armhf).

Fixes: gromox-2.14-41-g91c37ac94
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 466a51df1..60908bf8f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -19,7 +19,7 @@ saved_LDFLAGS="$LDFLAGS"
 saved_LIBS="$LIBS"
 AC_LANG_PUSH([C++])
 
-my_CPPFLAGS="-D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64"
+my_CPPFLAGS=""
 my_CFLAGS="-Wall -Wmissing-declarations -Wwrite-strings"
 my_CXXFLAGS="-Wall -Wmissing-declarations"
 my_LDFLAGS=""
-- 
2.55.0

